lib3h/src/dht/dht_event.rs

```diff
@@ -0,0 +1,74 @@
+use holochain_lib3h_protocol::Address;
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum DhtEvent {
+    /// We have received a gossip bundle from a remote peer,
+    /// pass it along to the dht backend for processing
+    RemoteGossipBundle(RemoteGossipBundleData),
+    /// Instructs implementors to send this binary gossip bundle
+    /// to the specified list of peerAddress' in a reliable manner.
+    GossipTo(GossipToData),
+    /// Instructs implementors to send this binary gossip bundle
+    /// to as many peers listed in peerList as possible.
+    /// It is okay if not all peers on the list receive the message.
+    UnreliableGossipTo(GossipToData),
+    /// Tell implementors that gossip is requesting we hold a peer discovery
+    /// data item. Note that this dht tracker has not actually marked this item
+    /// for holding until the implementors pass this event back in.
+    PeerHoldRequest(PeerHoldRequestData),
+    /// Tell implementors that gossip believes a peer has dropped
+    PeerTimedOut(String),
+    /// Tell implementors that gossip is requesting we hold a data item.
+    /// Note that this dht tracker has not actually marked this item
+    /// for holding until the implementors pass this event back in.
+    DataHoldRequest(DataHoldRequestData),
+    /// This dht tracker requires access to the data associated with a data hash.
+    /// This event should cause implementors to respond with a dataFetchResponse
+    /// event.
+    DataFetch(DataFetchData),
+    /// Response to a dataFetch event. Set `data` to `null` to indicate the
+    /// requested data is not available (it will be removed from gossip).
+    DataFetchResponse(DataFetchResponseData),
+    /// Tell our implementors that we are no longer tracking this data
+    /// locally. Implementors should purge this hash from storage,
+    /// but that can, of course, choose not to.
+    DataPrune(String),
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct RemoteGossipBundleData {
+    pub from_peer_address: String,
+    pub bundle: Vec<u8>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct GossipToData {
+    pub peer_address_list: Vec<String>,
+    pub bundle: Vec<u8>,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct PeerHoldRequestData {
+    pub peer_address: String,
+    pub transport: String,
+    pub data: Vec<u8>,
```

Comment:
Let's drop `data`... we don't have a specific use for it yet, and it may make more sense to add strongly-typed attributes rather than binary data.

Reply:
ok. I was thinking of changing it to `hint: String`, but removing it seems even better.

Reply:
I agree, type check is our best friend :)